charts/rancher-vsphere-csi/templates/node/daemonset.yaml

```diff
@@ -18,137 +18,128 @@ spec:
         app: vsphere-csi-node
         role: vsphere-csi
     spec:
+      nodeSelector:
+        kubernetes.io/os: linux
```

Comment:
Template this like [this](https://github.com/rancher/charts/blob/31f56d0b1c01592b70dd9763f1aadfed53e78572/charts/rancher-vsphere-csi/100.1.0%2Bup2.3.0/templates/controller/deployment.yaml#L20-L24), but for `csiNode` in values.yaml. Also, I do we need affinity/tolerations for rke and rke2 like we do for the deployment [here](https://github.com/rancher/charts/blob/31f56d0b1c01592b70dd9763f1aadfed53e78572/charts/rancher-vsphere-csi/100.1.0%2Bup2.3.0/templates/controller/deployment.yaml#L25-L62)? You may want to test that.

Reply:
Added the nodeSelector template